Felienne/hedy
# Makes sure to always generate grammar before starting tests

**Description**
In this PR we make a small improvement to the testing pipeline by first creating the grammar files before running all tests. This to ensure that all tests still pass in the case someone changing something in the grammar as well as the content without manually generating the grammar files.

**Fixes**
This PR fixes #2882 

**How to test**
GitHub should automatically take care of this but to validate you can manually run the `validate-tests` script in `build-tools/github` folder.
